pkg/gofr/gofr_test.go

```diff
@@ -130,3 +132,72 @@ func TestGofr_ServerRun(t *testing.T) {
 
 	resp.Body.Close()
 }
+
+func TestGofr_ServerHealthHandlerAddCheck(t *testing.T) {
+	var (
+		netClient = &http.Client{}
+		url       = "http://localhost:8000"
+		respn     response.Raw
+	)
+
+	g := New()
+	// Need to add a route to enable http server
+	g.GET("/", nil)
+	// Run the server
+	go g.Run()
+	time.Sleep(1 * time.Second)
+
+	// Send a GET request to the server
+	re, _ := http.NewRequestWithContext(context.Background(), http.MethodGet, url+"/.well-known/health", http.NoBody)
+	resp, err := netClient.Do(re)
+
+	// Assert that connection was successful
+	assert.NoError(t, err)
+
+	// make response bytes slice with content-length
+	responseBytes := make([]byte, resp.ContentLength)
+
+	// read the response into responseBytes
+	_, err = resp.Body.Read(responseBytes)
+	if err != nil && !errors.Is(err, io.EOF) {
+		t.Errorf("TEST failed: %v", err)
+
+		return
+	}
+
+	err = json.Unmarshal(responseBytes, &respn)
+	if err != nil {
+		t.Errorf("TEST failed %v", err)
+
+		return
+	}
+
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+	assert.Equal(t, response.Raw{Data: "OK"}, respn)
+
+	resp.Body.Close()
+}
+
+func TestGofr_ServerNotRunningForNoRoutes(t *testing.T) {
```

Comment:
I think this does not really confirms if the server is running or not. Even if g.Run call is delayed - it will still pass, even though server will eventually run.

Reply:
So we just need to add a delay to let the g.Run() call be processed and then, check if the HTTP request passes through!

Reply:
How do we know for sure how much time? I do not have solution, just asking. Can we check how others choose to test this scenario?